### PR TITLE
Clean and document 2D filter

### PR DESCRIPTION
**Pull request by dstansby**
This PR documents the 2D filtering code, cleans up un-used code, and refactors some of the remaining code. A bit of context:

- I've tried to give docstrings explaining what's happening in the critical places. Please be harsh when reviewing these!
- The whole of `base_tile_filter.py` was removed by moving `is_low_average()` to `tile_walker.py`. I think this makes the code much easier to follow.

### CODE REVIEW

File: src/cellfinder_core/detect/filters/plane/tile_walker.py [hunk 1]
```diff
@@ -1,69 +1,80 @@
...
-        self.ftf = OutOfBrainTileFilter(
-            out_of_brain_intensity_threshold=out_of_brain_threshold
-        )
+        corner_tile = img[0 : self.tile_width, 0 : self.tile_height]
+        corner_intensity = np.mean(corner_tile)
+        corner_sd = np.std(corner_tile)
+        # add 1 to ensure not 0, as disables
+        self.out_of_brain_threshold = (corner_intensity + (2 * corner_sd)) + 1
 
     def _get_tiles(self):  # WARNING: crops to integer steps
```
Comment (alessandrofelder) [inline src/cellfinder_core/detect/filters/plane/tile_walker.py:42]: should the `# WARNING: ` become part of the docstring?

File: src/cellfinder_core/detect/filters/plane/tile_walker.py [hunk 2]
```diff
@@ -1,69 +1,80 @@
...
+                tile = self.img[
                     x : x + self.tile_width, y : y + self.tile_height
                 ]
-                write_tile = self.thresholded_img[
-                    x : x + self.tile_width, y : y + self.tile_height
-                ]
-                yield x, y, read_tile, write_tile
+                yield x, y, tile
 
     def walk_out_of_brain_only(self):
```
Comment (alessandrofelder) [inline src/cellfinder_core/detect/filters/plane/tile_walker.py:57]: I agree with the docstring, I think the name of this function does not say what it does though?
```suggestion
    def mark_brain_tiles(self):
```
or
```suggestion
    def mark_bright_tiles(self):
```

**Review by alessandrofelder — CHANGES_REQUESTED** [review]
I like the simplification introduced by removing `base_tile_filter`.

* Minor comments on the docstrings 
* some suggestions to rename some variables to make the code more expressive and self-documenting while we're here (I decided this was in-scope for a documentation PR, but happy to be persuaded otherwise)?

File: src/cellfinder_core/detect/filters/plane/tile_walker.py [hunk 3]
```diff
@@ -1,69 +1,80 @@
...
 
-from cellfinder_core.detect.filters.plane.base_tile_filter import (
-    OutOfBrainTileFilter,
-    is_low_average,
-)
 
+class TileWalker:
+    """
+    A class to segment a 2D image into tiles, and mark each of the
+    tiles as good or bad depending on whether the average image
```
[the remark this answers is not in the document]
Comment (dstansby) [reply src/cellfinder_core/detect/filters/plane/tile_walker.py:10]: 👍 I like this. In `tile_walker.py` I've changed to using bring/dark terminology, and then in `plane_filter.py` I've renamed `good_tiles` to `inside_brain_tiles`, as that's what the bright tiles are being interpreted as at that point in the code.

File: src/cellfinder_core/detect/filters/plane/tile_walker.py (hunk 1 above)
Comment (dstansby) [reply src/cellfinder_core/detect/filters/plane/tile_walker.py:42]: I've taken a guess at what this warning means, and migrated it into the docstring

File: src/cellfinder_core/detect/filters/plane/tile_walker.py (hunk 2 above)
Comment (dstansby) [reply src/cellfinder_core/detect/filters/plane/tile_walker.py:57]: 👍 done

**Comment by dstansby** [comment]
Thanks for the comments, I think I've addressed all of them

**Review by alessandrofelder — APPROVED** [review]
Awesome!

**Pull request merged**